app/scripts/modules/core/whatsNew/whatsNew.read.service.ts

```diff
@@ -0,0 +1,58 @@
+import {module, IHttpService, ILogService, IPromise, IHttpPromiseCallbackArg} from 'angular';
+import {NetflixSettings} from 'netflix/netflix.settings';
+
+export interface IGistApiResponse {
+  // There are many other fields in the real response object.
+  files: {[fileKey: string]: IGistFile};
+  updated_at: string;
+}
+
+export interface IGistFile {
+  content: string;
+  filename?: string;
+  language?: string;
+  raw_url?: string;
+  size?: number;
+  truncated?: boolean;
+  type?: string;
+}
+
+export interface IWhatsNewContents {
+  contents: string;
+  lastUpdated: string;
+}
+
+export class WhatsNewReader {
+
+  private static extractFileContent(data: IGistApiResponse) {
+    return data.files[NetflixSettings.whatsNew.fileName].content;
+  }
+
+  static get $inject() { return ['$http', '$log']; }
+
+  constructor(private $http: IHttpService, private $log: ILogService) {}
+
+  public getWhatsNewContents(): IPromise<IWhatsNewContents> {
+    let gistId = NetflixSettings.whatsNew.gistId,
+      accessToken = NetflixSettings.whatsNew.accessToken || null,
+      url = ['https://api.github.com/gists/', gistId].join('');
+    if (accessToken) {
+      url += '?access_token=' + accessToken;
+    }
+    return this.$http.get(url)
+      .then((result: IHttpPromiseCallbackArg<IGistApiResponse>) => {
+        return {
+          contents: WhatsNewReader.extractFileContent(result.data),
+          lastUpdated: result.data.updated_at,
+        };
+      })
+      .catch((failure: IHttpPromiseCallbackArg<any>) => {
+        this.$log.warn('failed to retrieve gist for what\'s new dialog:', failure);
```

Comment:
could use a template string here to avoid having to escape the apostrophe in `what's` (I know you're copying from existing code, just pointing it out)